pilot/pkg/model/push_context_test.go

```diff
@@ -0,0 +1,248 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	networking "istio.io/api/networking/v1alpha3"
+)
+
+func createTestDestinationRuleConfig(name, namespace, host string) *Config {
+	return &Config{
+		ConfigMeta: ConfigMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: &networking.DestinationRule{
+			Host: host,
+			Subsets: []*networking.Subset{
+				{Name: "v1", Labels: map[string]string{"version": "v1"}},
+			},
+		},
+	}
+}
+
+func createTestDestionationRule(name Hostname, config *Config) map[Hostname]*Config {
+	return map[Hostname]*Config{name: config}
+}
+
+var (
+	pushContext1     = NewPushContext()
+	drConfig1        = createTestDestinationRuleConfig("test", "test-namespace", "test-host-1")
+	drConfig2        = createTestDestinationRuleConfig("test", "test-namespace", "test-host-2")
+	drConfig3        = createTestDestinationRuleConfig("test", "test-namespace", "test-host-3")
+	drConfig4        = createTestDestinationRuleConfig("test", "test-namespace", "test-host-4")
+	drConfig5        = createTestDestinationRuleConfig("test", "test-namespace", "test-host-5")
+	dr1              = createTestDestionationRule("test-host-1", drConfig1)
+	sidecarTypeProxy = &Proxy{
+		SidecarScope: &SidecarScope{
+			Config:           drConfig1,
+			destinationRules: dr1,
+		},
+		Type: SidecarProxy,
+	}
+
+	nonSidecarTypeProxy = &Proxy{
+		Type:            Router,
+		ConfigNamespace: "test-namespace",
+	}
+
+	pushContext2 = &PushContext{
+		allExportedDestRules: &processedDestRules{
+			hosts: []Hostname{"test-host-2"},
+			destRule: map[Hostname]*combinedDestinationRule{
+				"test-host-2": &combinedDestinationRule{
+					config: drConfig2,
+				},
+			},
+		},
+	}
+
+	pushContext3 = &PushContext{
```

Comment:
can you make these names more descriptive? No idea what `pushContext3` does

Reply:
Will adding comments suffice? I started adding description and its becoming too long